File: C++算法编程课C1/c++练习/OJ练习/4092.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;

    int ppb = 24 * 10 * 8;

    int tp = ppb * n;

    int ppm = tp / 2;

    int tfp = ppm / 8 * 3;
    int tfb = ppm / 8 / 10 * 5;
    int tfc = ppm / 8 / 10 / 24 * 7;

    int ts = tfp + tfb + tfc;

    int hours = ts / 3600;
    int rs = ts % 3600;
    int minutes = rs / 60;
    int seconds = rs % 60;

    cout << hours << "小时" << minutes << "分钟" << seconds << "秒" << endl;

    return 0;
}

File: C++算法编程课C1/c++练习/OJ练习/4092.py
n = int(input().strip())

ppb = 24 * 10 * 8

tp = ppb * n

ppm = tp / 2

tfp = ppm / 8 * 3
tfb = ppm / 8 / 10 * 5
tfc = ppm / 8 / 10 / 24 * 7

ts = tfp + tfb + tfc

hours = ts // 3600
rs = ts % 3600
minutes = rs // 60
seconds = rs % 60

print(f"{int(hours)}小时{int(minutes)}分钟{int(seconds)}秒")